codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1394,6 +1395,101 @@ getDependencyIdsForDependent dependent@(C.Reference.Id oid0 _) =
     isNotSelfReference (C.Reference.Id oid1 _) =
       oid0 /= oid1
 
+-- | Given two term (components) A and B, return the set of all terms that are along any "dependency path" from A to B,
+-- not including A nor B; i.e., the transitive dependencies of A that are transitive dependents of B.
+--
+-- For example, if A depends on X and Y, and Y depends on Z, and X and Z depend on B...
+--
+--     --X--
+--    /     \
+--   A       B
+--    \     /
+--     Y---Z
+--
+-- ...then `getDependenciesBetweenTerms A B` would return the set {X Y Z}
+getDependenciesBetweenTerms :: ObjectId -> ObjectId -> Transaction (Set ObjectId)
+getDependenciesBetweenTerms oid1 oid2 =
+  queryListCol sql (oid1, oid2) <&> Set.fromList
+  where
+    -- Given the example above, we'd have tables that look like this.
+    --
+    -- First, the `paths` table finds all paths from source `A` to sink `B`, exploring depth-first. As a minor
```

Comment:
finds all paths from source `A` (not just ones to `B`)
note: we can't know that a path doesn't end in `B` until we get there.